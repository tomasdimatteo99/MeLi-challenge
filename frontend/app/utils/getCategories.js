const axios = require('axios');
const { notFound } = require('next/navigation');

//Mover al backend.
const getCategories = async( id ) =>{
  try{
      const response  = await axios.get(`https://api.mercadolibre.com/categories/${id}`);
      const categories = response.data.path_from_root.map( catVal => catVal.name);
      return {categories: categories};
  }catch(err){
      notFound()
  }
}

module.exports = { getCategories };
