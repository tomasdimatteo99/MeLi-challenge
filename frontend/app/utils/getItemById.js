const axios = require('axios');
const { notFound } = require('next/navigation');

const getItemById = async( id ) =>{
  try{
      const response  = await axios.get(`http://localhost:8080/api/items/${id}`);
      return response.data;
  }catch(err){
      notFound()
  }
}

module.exports = { getItemById };