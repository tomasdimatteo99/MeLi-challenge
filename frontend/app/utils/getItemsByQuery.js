const axios = require('axios');
const { notFound } = require('next/navigation');

const getItemsByQuery = async(searchParams = "iphone") =>{
    try{
        const response  = await axios.get(`http://localhost:8080/api/items?q=${searchParams}`);
        return response.data;
    }catch(err){
        notFound()
    }
}

module.exports = { getItemsByQuery };