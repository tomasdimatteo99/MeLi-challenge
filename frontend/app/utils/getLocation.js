const axios = require('axios');
const { notFound } = require('next/navigation');

//Mover al backend.
const getLocation = async( id ) =>{
  try{
      const response  = await axios.get(`https://api.mercadolibre.com/items/${id}`);
      const location = response.data.seller_address.state.name;
      return {location: location};
  }catch(err){
      notFound()
  }
}

module.exports = { getLocation };
