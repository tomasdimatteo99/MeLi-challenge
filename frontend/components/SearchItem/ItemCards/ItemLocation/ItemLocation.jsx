import style from "./ItemLocation.module.scss";

export default function ItemLocation() {
  return (
    <div className={style.itemLocation}>
        <span>Capital Federal</span>
    </div>
  );
}