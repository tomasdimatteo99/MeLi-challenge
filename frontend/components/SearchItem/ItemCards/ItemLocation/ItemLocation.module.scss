@import "../../../../styles/sass/globals.scss";
.itemLocation {
    span {
        @include defaultFormat($fontSize: $fontSizeXXS, $textColor: $bgGray);
    }
    margin: 0 auto;
}
