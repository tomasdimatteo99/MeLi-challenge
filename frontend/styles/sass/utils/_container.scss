@use "../variable/mediaQueries" as mediaQuery;

//Setters
@mixin setWidth($width){
    max-width: $width;
    min-width: $width;
    width: $width;
}

@mixin setHeight($height){
    max-height: $height;
    min-height: $height;
    height: $height;
}

@mixin setDimension(
    $width,
    $height
){
    @include setWidth($width);
    @include setHeight($height);
}

@mixin setMargin(
    $margin:0,
    $margin-l:$margin,
    $margin-r:$margin,
    $margin-b:$margin,
    $margin-t:$margin
){
    margin-left: $margin-l;
    margin-right: $margin-r;
    margin-bottom: $margin-b;
    margin-top: $margin-t;
}

@mixin setMargin-x($margin-x){
    margin-left: $margin-x;
    margin-right: $margin-x;
}

@mixin setMargin-y($margin-y){
    margin-top: $margin-y;
    margin-bottom: $margin-y;
}

@mixin setPadding(
    $padding:0,
    $padding-l:$padding,
    $padding-r:$padding,
    $padding-b:$padding,
    $padding-t:$padding
){
    padding-left: $padding-l;
    padding-right: $padding-r;
    padding-bottom: $padding-b;
    padding-top: $padding-t;
}

@mixin setPadding-x($padding-x){
    padding-left: $padding-x;
    padding-right: $padding-x;
}

@mixin setPadding-y($padding-y){
    padding-top: $padding-y;
    padding-bottom: $padding-y;
}

//Modeling
@mixin container-fluid(
    $padding-x: 14px,
    $margin-x: auto,
    $width: 100vw,
    $height: auto
){
    @include setMargin-x($margin-x);
    @include setPadding-x($margin-x);
    @include setWidth($width);
    @include setHeight($height);

    //Small
    @media #{mediaQuery.$sm} {
        @include setMargin-x($margin-x);
        @include setPadding-x($margin-x);
        @include setWidth($sm-px);
    }
    //Medium
    @media #{mediaQuery.$md} {
        @include setMargin-x($margin-x);
        @include setPadding-x($margin-x);
        @include setWidth($md-px);
    }
    //Large
    @media #{mediaQuery.$lg} {
        @include setMargin-x($margin-x);
        @include setPadding-x($margin-x);
        @include setWidth($lg-px);
    }
    //Extra - Large
    @media #{mediaQuery.$xl} {
        @include setMargin-x($margin-x);
        @include setPadding-x($margin-x);
        @include setWidth($xl-px);
    }
    //Extra extra - Large
    @media #{mediaQuery.$xxl} {
        @include setMargin-x($margin-x);
        @include setPadding-x($margin-x);
        @include setWidth($xxl-px);
    }
}

@mixin container(
    $padding-x: 14px,
    $margin-x: auto,
    $width: 80vw,
    $height: auto
){
    @include container-fluid(
        $padding-x,
        $margin-x,
        $width,
        $height
    );
}

@mixin simpleContainer(
    $padding-x: 14px,
    $margin-x: auto,
    $width: 100vw,
    $height: auto
){
    @include setMargin-x($margin-x);
    @include setPadding-x($margin-x);
    @include setWidth($width);
    @include setHeight($height);
}