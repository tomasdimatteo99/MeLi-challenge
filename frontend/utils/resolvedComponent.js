const resolvedComponent = async (Component, props) => {
    const ComponentResolved = await Component(props);
    return () => ComponentResolved;
}

module.exports = { resolvedComponent };