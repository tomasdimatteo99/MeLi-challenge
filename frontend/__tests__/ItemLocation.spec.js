import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import ItemLocation from "@/components/SearchItem/ItemCards/ItemLocation/ItemLocation";
import { resolvedComponent } from "@/utils/resolvedComponent";

describe("ItemLocation", () => {
  it("should render the component correctly", async () => {
    const ItemLocationResolved = await resolvedComponent(ItemLocation, {
        "id": "MLA1397418693"
    });
    const { container } = render(<ItemLocationResolved/>);
    expect( container ).toMatchSnapshot();
  });
});


