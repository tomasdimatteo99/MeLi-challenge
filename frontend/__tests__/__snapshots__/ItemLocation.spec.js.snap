// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemLocation should render the component correctly 1`] = `
<div>
  <div
    class="itemLocation"
  >
    <h3>
      Capital Federal
    </h3>
  </div>
</div>
`;
